lib/classes/PluginManager.js

```diff
@@ -90,8 +90,10 @@ class PluginManager {
     this.cliCommands = commands;
   }
 
-  addPlugin(Plugin) {
+  addPlugin(Plugin, isExternal = false) {
     const pluginInstance = new Plugin(this.serverless, this.cliOptions);
+    // isExternal differentiates plugin as OOTB or not
+    Object.assign(pluginInstance, { isExternal });
```

Comment:
It'll be more transparent to create `this.externalPlugins` collection (as set instance), and just group all the plugins there.

Assigning a property to a plugin instance, while should in most cases be harmless, it's destructive, note that we're interferring into plugin API, we can't be sure if we won't break things

Reply:
u mean keeping `this.plugins `as it is and adding `this.externalPlugins`, right ?

Reply:
Yes, exactly